File: 2026/07-July/24-Friday/solution.cs
using System.Collections.Generic;

public class Solution
{
    public int UniqueXorTriplets(int[] nums)
    {
        int n = nums.Length;

        if (n == 1)
            return 1;

        HashSet<int> pairXors = new HashSet<int>();

        for (int i = 0; i < n; i++)
        {
            for (int j = i + 1; j < n; j++)
            {
                pairXors.Add(nums[i] ^ nums[j]);
            }
        }

        HashSet<int> tripletXors = new HashSet<int>();

        foreach (int px in pairXors)
        {
            foreach (int value in nums)
            {
                tripletXors.Add(px ^ value);
            }
        }

        return tripletXors.Count;
    }
}

File: 2026/07-July/24-Friday/solution.py
from typing import List

class Solution:
    def uniqueXorTriplets(self, nums: List[int]) -> int:
        n = len(nums)
        if n == 1:
            return 1

        pair_xors = set()

        for i in range(n):
            x = nums[i]
            for j in range(i + 1, n):
                pair_xors.add(x ^ nums[j])

        triplet_xors = set()

        for px in pair_xors:
            for v in nums:
                triplet_xors.add(px ^ v)

        return len(triplet_xors)
